integration/hdfs/Dockerfile

```diff
@@ -15,63 +15,34 @@
 # specific language governing permissions and limitations
 # under the License.
 
-FROM gelog/hadoop
+FROM arrow:python-3.6
 
-RUN apt-get update && \
-    apt-get install -y \
-        autoconf \
-        automake \
-        make \
-        gcc \
-        g++ \
-        git \
-        wget \
-        pkg-config \
-        ninja-build
-
-ENV CC=gcc \
-    CXX=g++ \
-    PATH=/opt/conda/bin:$PATH \
-    CONDA_PREFIX=/opt/conda
-
-# install dependencies
-ARG PYTHON_VERSION=3.6
-ADD ci/docker_install_conda.sh \
-    ci/conda_env_cpp.yml \
-    ci/conda_env_python.yml \
-    /arrow/ci/
-RUN arrow/ci/docker_install_conda.sh && \
-    conda install -c conda-forge \
-        --file arrow/ci/conda_env_cpp.yml \
-        --file arrow/ci/conda_env_python.yml \
-        python=$PYTHON_VERSION && \
-    conda clean --all
-
-# installing in the previous step boost=1.60 and boost-cpp=1.67 gets installed,
-# cmake finds 1.60 and parquet fails to compile
-# installing it in a separate step, boost=1.60 and boost-cpp=1.64 gets
-# installed, cmake finds 1.64
-# libhdfs3 needs to be pinned, see ARROW-1465 and ARROW-1445
+# installing libhdfs3, it needs to be pinned, see ARROW-1465 and ARROW-1445
 RUN conda install -y -c conda-forge hdfs3 libhdfs3=2.2.31 && \
     conda clean --all
 
+# installing libhdfs (JNI)
+ARG HADOOP_VERSION=2.6.0
+ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 \
+    HADOOP_HOME=/usr/local/hadoop \
+    HADOOP_OPTS=-Djava.library.path=/usr/local/hadoop/lib/native \
+    PATH=$PATH:$HADOOP_HOME/bin:$HADOOP_HOME/sbin
+RUN apt-get update -y && \
+    apt-get install -y openjdk-8-jdk && \
+    wget http://archive.apache.org/dist/hadoop/core/hadoop-$HADOOP_VERSION/hadoop-$HADOOP_VERSION.tar.gz && \
```

Comment:
We aren't supposed to be using the dist system like this. Is there another way to download the libraries that either uses the Apache mirror system, or to get them from a package manager?

Reply:
Right, let me take a look.

Reply:
Updated